### PR TITLE
fix: linter & dependency issues on tox.ini template

### PR DESCRIPTION
**Pull request by jneo8**
Fix couple issues on tox.ini:

- Options to disable pylint & mypy on tox.ini, default disable.
- Option to include requirements.txt in root directory.
- Include testenv deps only if test is not none.

### CODE REVIEW

**Review by chanchiwai-ray — APPROVED** [review]
I checked action output for snap-tempest and logroated, and they seems okay

File: terraform-plans/templates/github/tox.ini.tftpl [hunk 1]
```diff
@@ -22,9 +22,12 @@ commands =
     black --check --diff --color .
     isort --check --diff --color .
     flake8
+%{ if enable_pylint == "yes" ~}
     pylint --recursive=y .
-    # mypy temporarily disabled because it doesn't gracefully recursively find files like the others
-    #mypy --install-types --non-interactive .
+%{ endif ~}
+%{ if enable_mypy == "yes" ~}
+    mypy --install-types --non-interactive .
```
Comment (samuelallan72) [inline terraform-plans/templates/github/tox.ini.tftpl:29]: It's checking for "yes" here, but in the variables it's setting "true" or "false"

**Review by samuelallan72 — CHANGES_REQUESTED** [review]

File: terraform-plans/templates/github/tox.ini.tftpl (hunk 1 above)
Comment (jneo8) [reply terraform-plans/templates/github/tox.ini.tftpl:29]: Ah, I forget change it to true. Thanks for find it out.

**Review by samuelallan72 — APPROVED** [review]

**Review by chanchiwai-ray — APPROVED** [review]
